### DISCUSSION TITLE
BT-161 and framework agreements

### DISCUSSION THREAD

**Discussion by EEanalyst** [OP-TED/eForms-SDK#413] [Q&A] [answered]
As we were adding framework agreement fields to the XML we ran into couple of problems:
1. According to the documentation BT-709 and BT-660 are optional, so we didn't add them but when we added BT-118 to the XML we got a validation error:

"For any Result Notice without Group Framework Maximum Value (BT-156), and for which all Framework Maximum Value (BT-709) share a same currency, the Notice Framework Maximum Value (BT-118) value must be equal to the sum of Framework Maximum Value (BT-709) or to the sum of Framework Estimated Value (BT-660)"

Why did we get this error if we don't have anything related to BT-709 or BT-660 in XML? 

2. To remove errors we decided to remove BT-118 and only use BT-161. This solution no longer gave us validation errors, but would there be validation errors in later versions? We are currently on version 1.3. 
Is BT-161 even allowed when a framework agreement is involved?

**Comment by YvesJo** [comment, marked as answer]
Indeed, in later versions there will also be an error as BT-118 is about Framework Agreement and in absence of Framework Agreements, this information should not exist. In a later version (most likely SDK 1.8) a condition will be applied to prevent the specification of BT-118 when no awarded lot with a Framework Agreement exist.
BT-161 is about "execution" contracts (i.e. engaged amounts) and having Framework Agreements in a notice does not prevent the existence of "execution" contracts (a notice may cover various situations with some lot results being about execution contracts, inside or not a Framework contract, and others about framework contracts), so yes BT-161 is also allowed when the notice contains Framework Agreements.
